pkg/proxy/userspace/proxier.go

```diff
@@ -294,6 +296,7 @@ func (proxier *Proxier) addServiceOnPort(service proxy.ServicePortName, protocol
 		return nil, err
 	}
 	si := &serviceInfo{
+		isAlive:             1,
```

Comment:
at struct creation time, are all the the field values in the struct guaranteed to be visible on all threads ?
The Go memory model didnt seem to call it out explicitly.(unless I missed it )
This also impacts other fields in the struct.
Otherwise, looks good to me.